controllers/storagecluster/cephcluster.go

```diff
@@ -170,6 +177,15 @@ func (obj *ocsCephCluster) ensureCreated(r *StorageClusterReconciler, sc *ocsv1.
 	if err != nil {
 		if errors.IsNotFound(err) {
 			if sc.Spec.ExternalStorage.Enable {
+				// External Mode and there is no existing CephCluster,
+				// verify the monitoring endpoints only for the first time
+				err := verifyMonitoringEndpointsOfCephCluster(cephCluster, r.Log)
+				if err != nil {
+					r.Log.Error(err, "Could not connect to the Monitoring Endpoints.",
+						"CephCluster", klog.KRef(sc.Namespace, sc.Name))
+					return reconcile.Result{}, err
+				}
+				r.Log.Info("Monitoring Information found. Monitoring will be enabled on the external cluster.", "CephCluster", klog.KRef(sc.Namespace, sc.Name))
```

Comment:
Can we just keep this code in the original location and just move up L175-L176 before that code section? 
It can also solve the issue In my last comment and will aggregate all checks in a single locations (right now you have 2 sections that take the same underlaying assumption - external mode)

Reply:
We can definitely move the checking of an existing CephCluster, ie L175 and 176 up. But I was trying to avoid this in an effort to not to duplicate the same code (for found logic) in two places.
As I explained previously, the above scenario (of overriding the configurations of an existing CephCluster) will never happen.
Let me know WDYT.